include/deal.II/sundials/n_vector.templates.h

```diff
@@ -487,7 +492,7 @@ template <typename VectorType,
           std::enable_if_t<is_serial_vector<VectorType>::value, int>>
 MPI_Comm SUNDIALS::internal::NVectorOperations::get_communicator(N_Vector)
 {
-  return nullptr;
+  return {};
```

Comment:
So this returns a default initialized value of whatever type MPI_Comm is. It would be an int (?) if MPI is not included but that doesn't matter due to the new #ifndef? Makes sense I guess. Thanks again!

Reply:
Yes, exactly. We return `nullptr` in `get_communicator_as_void_ptr` if deal.II is built without MPI support or if a serial vector is provided. This seems to match the intent of the code you have written.

Reply:
I think this should simply be `MPI_COMM_SELF` if you have nothing better to return. We define this in `base/mpi.h` even if we don't have MPI:
```
#if !defined(DEAL_II_WITH_MPI) && !defined(DEAL_II_WITH_PETSC)
// without MPI, we would still like to use
// some constructs with MPI data
// types. Therefore, create some dummies
using MPI_Comm     = int;
using MPI_Request  = int;
using MPI_Datatype = int;
using MPI_Op       = int;
#  ifndef MPI_COMM_WORLD
#    define MPI_COMM_WORLD 0
#  endif
#  ifndef MPI_COMM_SELF
#    define MPI_COMM_SELF 0
#  endif
```

Reply:
Perfect! That's what SUNDIALS expects of us;)